dashboard/app/controllers/experiments_controller.rb

```diff
@@ -33,6 +33,11 @@ def set_single_user_experiment
       return
     end
 
+    if Experiment.enabled?(experiment_name: experiment_name, user: current_user)
+      redirect_to '/', flash: {alert: "Already enabled in experiment '#{params[:experiment_name]}'."}
+      return
+    end
```

Comment:
to be resilient to any existing bad data, you could also use `.where(...).destroy_all` instead of `.find_by(...).destroy` below

Reply:
Great suggestion - will follow up.